third_party/docfx-doclet-143274/src/main/java/com/microsoft/build/YmlFilesBuilder.java

```diff
@@ -143,14 +147,22 @@ void process() {
                   .filter(pkg -> !packageLookup.isApiVersionStubPackage(pkg))
                   .collect(Collectors.toList()));
 
-      // Get recommended ApiVersion for new Library Overview
-      HashSet<ApiVersion> versions = new HashSet<>();
+      // Calculate the recommended package based on the latest stable Version ID. This will be
+      // overridden by the recommended_package in the RepoMetadata, if set
+      HashMap<ApiVersion, String> packageVersions = new HashMap<>();
       for (PackageElement pkg : allPackages) {
-        packageLookup.extractApiVersion(pkg).ifPresent(versions::add);
+        Optional<ApiVersion> apiVersion = packageLookup.extractApiVersion(pkg);
+        apiVersion.ifPresent(
+            version -> packageVersions.put(version, String.valueOf(pkg.getQualifiedName())));

```

Comment:
(very) nit: I'm not sure precalculating like this is necessary.

```java
// Get recommended ApiVersion for new Library Overview
HashSet<ApiVersion> versions = new HashSet<>();
for (PackageElement pkg : allPackages) {
  packageLookup.extractApiVersion(pkg).ifPresent(versions::add);
}
...
String recommendedPackage = repoMetadata.getRecommendedPackage()
    .orElseGet(() -> {
        if (packageVersions.isEmpty()) return "";

        ApiVersion recommended = ApiVersion.getRecommended(packageVersions.keySet());
        return String.valueOf(recommended.getQualifiedName());
    });
```

Reply:
makes sense! i've updated